cpp/CMakeLists.txt

```diff
@@ -47,6 +47,12 @@ option(CUDF_ENABLE_ARROW_ORC "Build the Arrow ORC adapter" OFF)
 option(CUDF_ENABLE_ARROW_PYTHON "Find (or build) Arrow with Python support" OFF)
 option(CUDF_ENABLE_ARROW_PARQUET "Find (or build) Arrow with Parquet support" OFF)
 option(CUDF_ENABLE_ARROW_S3 "Build/Enable AWS S3 Arrow filesystem support" ON)
+option(
+  CUDF_USE_PER_THREAD_DEFAULT_STREAM
+  "Build cuDF with per-thread default stream, including passing the per-thread default
+         stream to external libraries."
+  OFF
+)
 option(PER_THREAD_DEFAULT_STREAM "Build with per-thread default stream" OFF)
```

Comment:
We should update the option comment here to note that this flag is deprecated.
We should also mark this as advanced https://cmake.org/cmake/help/latest/command/mark_as_advanced.html so the option doesn't show in ccmake or cmake-gui